src/sentry/integrations/vsts/webhooks.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import absolute_import
+from .client import VstsApiClient
+
+from sentry.models import Identity, Integration, sync_group_assignee_inbound
+from sentry.api.base import Endpoint
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import csrf_exempt
+
+import re
+UNSET = object()
+# Pull email from the string: u'lauryn <lauryn@sentry.io>'
+EMAIL_PARSER = re.compile(r'<(.*)>')
+
+
+class WorkItemWebhook(Endpoint):
+    authentication_classes = ()
+    permission_classes = ()
```

Comment:
We should be doing some sort of Authentication + Authorization for this endpoint. As far as I can tell, the way it is now allows you alter anyone's Issues by replicating a VSTS webhook request. Unless I'm missing something?

Reply:
I'm relatively sure you're correct. The model I'm using for this is @macqueen Jira code: https://github.com/getsentry/sentry/blob/master/src/sentry/integrations/jira/webhooks.py#L11:L13

Looks like she verifies it with a token that's given. https://github.com/getsentry/sentry/blob/master/src/sentry/integrations/jira/webhooks.py#L20:L23
Hmmm.... I wonder what I can do for vsts... I'm not sure there is an equivalent.

Reply:
You can specify Basic Auth creds (or any Headers) through the UI, so there's definitely a way to do it via the API.

Looks like Basic Auth is via these attributes:

![image](https://user-images.githubusercontent.com/36059/42401615-501db028-812b-11e8-8b35-77020d6702d4.png)

and Headers being similar (the little arrow is a newline):

![image](https://user-images.githubusercontent.com/36059/42401653-85d917ca-812b-11e8-99c0-934e6985b2d5.png)

We can probably leverage one of these with some shared secret.

Reply:
Ok I'll check that out! Thanks. :)